packages/fungible-token/bridge-fungible-token/proxy/src/proxy.sw

```diff
@@ -0,0 +1,74 @@
+contract;
+
+use std::{
+    call_frames::first_param,
+    execution::run_external,
+    hash::{sha256, Hash},
+    string::String,
+};
+use standards::{src14::SRC14, src20::SRC20, src5::{AccessError, SRC5, State}};
+use contract_message_receiver::MessageReceiver;
+use interface::{bridge::Bridge, src7::{Metadata, SRC7}};
+
+abi Proxy {
+    #[storage(read)]
+    fn _proxy_owner() -> State;
+
+    #[storage(read)]
+    fn _proxy_target() -> ContractId;
+
+    #[storage(read, write)]
+    fn _proxy_change_owner(new_owner: Identity);
+}
+
+#[namespace(SRC14)]
+storage {
+    // target is at sha256("storage_SRC14_0")
+    target: ContractId = ContractId::zero(),
+    // owner is at sha256("storage_SRC14_1")
+    owner: State = State::Uninitialized,
+}
+
+impl SRC14 for Contract {
+    #[storage(write)]
+    fn set_proxy_target(new_target: ContractId) {
+        only_owner();
+        storage.target.write(new_target);
+    }
+}
+
+#[fallback]
+#[storage(read)]
+fn fallback() {
+    // pass through any other method call to the target
+    run_external(storage.target.read())
+}
+
+#[storage(read)]
+fn only_owner() {
+    require(
+        storage
+            .owner
+            .read() == State::Initialized(msg_sender().unwrap()),
+        AccessError::NotOwner,
+    );
+}
+
+impl Proxy for Contract {
+
+    #[storage(read)]
+    fn _proxy_owner() -> State {
+        storage.owner.read()
+    }
+
+    #[storage(read)]
+    fn _proxy_target() -> ContractId {
+        storage.target.read()
+    }
+
+    #[storage(read, write)]
+    fn _proxy_change_owner(new_owner: Identity) {
+        only_owner();
```

Comment:
As the `storage.owner` starts as `State::Uninitialized`; a new owner can never be set as this `only_owner()` will always revert.
Instead either:
- `storage.owner` can start as `State::Initialized` as in: https://github.com/FuelLabs/sway-standards/blob/master/examples/src5-ownership/initialized_example/src/initialized_example.sw
- an `initialize_ownership` method can be included as in: https://github.com/FuelLabs/sway-libs/blob/0f47d33d6e5da25f782fc117d4be15b7b12d291b/libs/src/ownership.sw#L127

Reply:
@SwayStar123 wrote the storage initialization and I packed it up here:

https://github.com/FuelLabs/fuel-bridge/blob/ae2ec304463e2fd5f903440d8fc562346148d551/packages/fungible-token/bridge-fungible-token/tests/utils/setup.rs#L590

I am not sure if there is a best practice defined. I would have gone the storage.unwrap_or(configurable) path if I had implemented it because I didn't know you can set storage slots on deployment.

Reply:
I believe i set it so the owner is set at deployment (with the storage slots changed via the sdk). If this is not best practice feel free to change it.
It is not possible to use a configurable in storage hence why i did this